File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/converter/PassThroughConverter.java
package com.roxiemobile.networkingapi.network.rest.converter;

import com.roxiemobile.networkingapi.network.http.MediaType;
import com.roxiemobile.networkingapi.network.rest.response.ResponseEntity;
import com.roxiemobile.networkingapi.network.rest.response.error.nested.ConversionException;

import org.jetbrains.annotations.NotNull;

public final class PassThroughConverter extends AbstractCallResultConverter<byte[]> {

// MARK: - Methods

    @Override
    public @NotNull ResponseEntity<byte[]> convert(@NotNull ResponseEntity<byte[]> entity) throws ConversionException {
        return entity;
    }

    @Override
    protected @NotNull MediaType[] supportedMediaTypes() {
        return SUPPORTED_MEDIA_TYPES;
    }

// MARK: - Constants

    private static final @NotNull MediaType[] SUPPORTED_MEDIA_TYPES = new MediaType[]{
            MediaType.ALL
    };
}

File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/converter/PassThroughConverter.kt
@file:Suppress("unused")

package com.roxiemobile.networkingapi.network.rest.converter

import com.roxiemobile.networkingapi.network.http.MediaType
import com.roxiemobile.networkingapi.network.rest.response.ResponseEntity
import com.roxiemobile.networkingapi.network.rest.response.error.nested.ConversionException

class PassThroughConverter: AbstractCallResultConverter<ByteArray>() {

// MARK: - Methods

    @Throws(ConversionException::class)
    override fun convert(responseEntity: ResponseEntity<ByteArray>): ResponseEntity<ByteArray> {
        return responseEntity
    }

    override fun supportedMediaTypes(): Array<MediaType> {
        return SUPPORTED_MEDIA_TYPES
    }

// MARK: - Companion

    companion object {

        private val SUPPORTED_MEDIA_TYPES = arrayOf(
            MediaType.ALL,
        )
    }
}
